src/agent/src/rpc.rs

```diff
@@ -639,8 +669,8 @@ impl protocols::agent_ttrpc::AgentService for AgentService {
 
         let resp = Empty::new();
 
-        if res.is_some() {
-            let oci_res = rustjail::resources_grpc_to_oci(&res.unwrap());
+        if let Some(res) = res.as_ref() {
```

Comment:
That's the way!  There are a bunch of places elsewhere in this patch (and elsewhere in the code in general) where this construct is what you want.